src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1393,6 +1393,123 @@ xclLoadXclBin(const xclBin *buffer)
   return ret;
 }
 
+/*
+ * xclRegisterAxlf()
+ */
+int shim::xclRegisterAxlf(const axlf *buffer)
```

Comment:
This looks a lot like xclLoadAxlf()?  Should we refactor and share code?

Reply:
Yes. You are correct. Actually I don't want to disturb the existing code. Hence added extra function for registration. 
Anyway I will cleanup once the changes are stable